front/src/views/leisure/leisure.view.js

```diff
@@ -15,32 +16,175 @@ export default {
+      this.autocomplete.addListener('place_changed', this.onPlaceChanged);
+
+      navigator.geolocation.getCurrentPosition(this.setMapCenter);
+    },
+    setMapCenter (position) {
+      if (position) {
+        const pos = {
+          lat: position.coords.latitude,
+          lng: position.coords.longitude
+        };
+        this.mapCenter = pos;
+        this.map.setCenter(this.mapCenter);
+      } else {
+        this.map.setCenter(this.mapCenter);
+      }
+      this.getNearbyPlaces(this.mapCenter);
+    },
+    onPlaceChanged () {
+      const place = this.autocomplete.getPlace(),
+            position = {
+              lat: place.geometry.location.lat(),
+              lng: place.geometry.location.lng()
+            };
+      if (place.geometry) {
+        this.map.panTo(place.geometry.location);
+        this.map.setZoom(15);
+      }
+      this.getNearbyPlaces(position);
+    },
+    async getNearbyPlaces (position) {
+      const search = {
+        bounds: this.map.getBounds(),
+        types: this.leisureTypes,
+        location: position,
+        radius: this.radius * 1000,
+        photo: null
+      };
+
+      this.placesService.nearbySearch(search, (results, status, pagination) =>
+        this.nearbySearchCallback(results, status, pagination));
+    },
+    nearbySearchCallback (places, status, pagination) {
+      this.clearMarkers();
+      this.places = [];
+      if (status === this.google.maps.places.PlacesServiceStatus.OK) {
+        places.forEach((place, i) => {
+          this.markers[i] = new this.google.maps.Marker({
+            position: place.geometry.location,
+            animation: this.google.maps.Animation.DROP
+          });
+
+          if (place.photos) {
+            place.photos = place.photos[0].getUrl();
+          }
+          if (place.opening_hours) {
+            if (place.opening_hours.open_now) {
+              place.opening_hours = this.$t('leisure.placeOpenState');
+            } else {
+              place.opening_hours = this.$t('leisure.placeCloseState');
+            }
+          }
+
+          this.getNextPage = pagination.hasNextPage && function () {
+            pagination.nextPage();
+          };
+
+          this.markers[i].placeResult = place;
+          setTimeout(this.dropMarker(i), i * 100);
+        });
+        this.places = places;
+        console.log(places);
+      }
+    },
+    dropMarker (i) {
+      const markers = this.markers,
+            map = this.map;
+      return function () {
+        markers[i].setMap(map);
+      };
+    },
+    clearMarkers () {
+      for (let i = 0; i < this.markers.length; i++) {
+        if (this.markers[i]) {
+          this.markers[i].setMap(null);
+        }
+      }
+      this.markers = [];
+    },
+    nextPage () {
+      if (this.getNextPage) {
+        this.getNextPage();
+      }
+    }
+  },
+  watch: {
+    showMap (value) {
+      if (value) {
+        this.initializeMap();
+      }
+    }
+  },
```

Comment:
do we realy need watcher? might we can use just computed property or other ways?